### ISSUE TITLE
Multibind raise error with classes with constructor parameters

### ISSUE THREAD

**Issue by BrunoBiluca** [alecthomas/injector#153]
Hello guys,

I have a S3 class that I use to connect to some S3 Buckets, because of that I need to use multibind to use many buckets in one resource.

My S3 connection classe is like that:
```python
import boto3


class S3:
    def __init__(self, aws_access_key, aws_secret_key, bucket, region_name, *args, **kwargs):
        self.aws_secret_key = aws_secret_key
        self.aws_access_key = aws_access_key
        self.bucket_name = bucket
        self.region_name = region_name
```

My declarative container is like that:

```python
class EnrichmentContainer(DeclarativeContainer):
    def configure(self, binder: Binder):
        config = get_config()

        binder.multibind(
            Dict[str, S3],
            to={
                "enrich_data_s3_bucket": S3(config.AWS_ACCESS_KEY, config.AWS_SECRET_KEY, config.ENRICH_DATA_S3_BUCKET, config.S3_REGION_NAME),
                "public_uploaded_files_s3_bucket": S3(
                    config.AWS_ACCESS_KEY,
                    config.AWS_SECRET_KEY,
                    config.PUBLIC_UPLOADED_FILES_BUCKET,
                    config.PUBLIC_UPLOADED_FILES_BUCKET_S3_REGION_NAME
                )
            },
            scope=request
        )
```

and in my Resource I'm injecting the dependencies like that:

```python
class EnrichmentDataResource(Resource):

    @inject
    def __init__(self,
                 enrichment_repository: EnrichmentRepository,
                 s3_buckets: Dict[str, S3],
                 *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.enrichment_repository = enrichment_repository
        self.s3_buckets = s3_buckets
```

But I'm getting the following error, and I tried all the providers to construct this object for me:

```
Traceback (most recent call last):
  File "C:\Users\bruno\Documents\projects\ct-enriched-data-api\venv\Lib\site-packages\injector\__init__.py", line 1015, in call_with_injection
    return callable(*full_args, **dependencies)
TypeError: __init__() missing 4 required positional arguments: 'aws_access_key', 'aws_secret_key', 'bucket', and 'region_name'
```

Please some bright soul that can enlight me about this, thanks for the help.

**Comment by jstasiak** [comment]
Use the public documented API only (see the examples in the readme and [the API reference](https://injector.readthedocs.io/en/latest/api.html)) – `Binder.create_binding()` is private and doesn't do what I think you believe it means, you want to use `Binder.bind()` (or in this case `Binder.multibind()`) instead.
 
If this doesn't help please provide *full* stack trace, your Python version, Injector version and the code used to initialize `Injector`.

**Comment by BrunoBiluca** [comment]
Hi, I edited my question and change the `Binder.create_binding()` to `Binder.multibind()`, this was I typing mistake of mine.

My problem was not related to the multibind, but instead there was a class that dependend on the simple bind S3 class, so when I changed the `binder.bind(s3)` to `multibind(Dict[str], S3)` it lose the parameters to construct the S3 class in the class that needed only one S3 parameter.

Thanks you the support it helped me to think out of the box.
